backend/src/org/akvo/lumen/endpoint/flow.clj

```diff
@@ -1,44 +1,18 @@
 (ns org.akvo.lumen.endpoint.flow
-  (:require [clojure.string :as str]
-            [clojure.set :as set]
-            [compojure.core :refer :all]
-            [ring.util.response :refer (response)]
-            [hugsql.core :as hugsql]
-            [org.akvo.lumen.import.flow :as flow-import]))
+  (:require [compojure.core :refer :all]
+            [org.akvo.lumen.lib.flow :as flow]
+            [ring.util.response :refer [response]]))
 
 
-(hugsql/def-db-fns "org/akvo/lumen/import/flow.sql")
-
-(defn remove-empty-folders
-  "Remove empty folders"
-  [folders-and-surveys]
-  (loop [fas folders-and-surveys]
-    (let [folder-ids (set (map :folderId fas))
-          next-fas (remove #(and (= (:type %) "folder")
-                                 (not (folder-ids (:id %))))
-                           fas)]
-      (if (= (count fas) (count next-fas))
-        fas
-        (recur next-fas)))))
-
 (defn endpoint [{{:keys [flow-report-database-url]} :config}]
   (context "/api/flow" _
     (GET "/folders-and-surveys/:org-id" request
-      (let [org-id (-> request :params :org-id)
-            root-ids (flow-import/root-ids org-id (:jwt-claims request))]
-        (if-not (empty? root-ids)
-          (response (remove-empty-folders
-                     (descendant-folders-and-surveys-by-folder-id
-                      (format flow-report-database-url org-id)
-                      {:folder-ids root-ids}
-                      {}
-                      {:identifiers identity})))
+      (let [items (flow/folders-and-surveys (:jwt-claims request)
+                                            flow-report-database-url
+                                            (-> request :params :org-id))]
+        (if items
+          (response items)
           (response ()))))
```

Comment:
@kardan this is outside the change set, so probably ignore for now :)

But I think every endpoint/domain that returns a collection should never return 404/`nil` as this one apparently does. Much better IMO to just return an empty collection.